### PR TITLE
New data source: `azurerm_databricks_workspace`

### LINKED ISSUE #8475: Support for Databricks Workspace Data Source
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

There is currently support for creating azure databricks workspaces using the `azurerm_databricks_workspace` resource, but there is no way to reference them as a data source. A data source for databricks workspaces would enable these resources to be leveraged by Terraform templates that do not create the workspace.

### New Data Source

* `data azurerm_databricks_workspace`

### Potential Terraform Configuration

```hcl
data "azurerm_databricks_workspace" "example" {
  name                = "myworkspace"
  resource_group_name = "some-resource-group"
}

output "workspace_url" {
  value = data.azurerm_databricks_workspace.example.workspace_url
}
output "workspace_id" {
  value = data.azurerm_databricks_workspace.example.workspace_id
}
```
Exports:
- Resource Id
- Workspace Url
- Workspace Id
- Managed Resource Group Id
- Etc.

### References

See existing `azurerm_databricks_workspace` [resource documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/databricks_workspace).

### PR DESCRIPTION
**Pull request by favoretti**
Fixes #8475

### CODE REVIEW

File: azurerm/internal/services/databricks/databricks_workspace_data_source.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+		return fmt.Errorf("Error making Read request on EventHub %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("workspace_id", resp.WorkspaceID)
+	d.Set("workspace_url", resp.WorkspaceURL)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/databricks/databricks_workspace_data_source.go:65]: can we pull these from `resp.Properties` instead?

File: azurerm/internal/services/databricks/tests/databricks_workspace_data_source_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-databricks-%[1]d"
+  location = "%[2]s"
+}
+
+data "azurerm_databricks_workspace" "example" {
+  name                = "example-workspace"
```
Comment (tombuildsstuff) [inline azurerm/internal/services/databricks/tests/databricks_workspace_data_source_test.go:40]: we need to create/reference an existing workspace here :)

File: azurerm/internal/services/databricks/tests/databricks_workspace_data_source_test.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  resource_group_name = "azurerm_resource_group.test.name"
```
Comment (tombuildsstuff) [inline azurerm/internal/services/databricks/tests/databricks_workspace_data_source_test.go:41]:
```suggestion
  resource_group_name = azurerm_resource_group.test.name
```

File: website/docs/d/databricks_workspace.html.markdown [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+## Argument Reference
+
+* `name` - The name of the Databricks Workspace.
+* `resource_group_name` - The Name of the Resource Group where the Databricks Workspace exists.
+
+## Attributes Reference
+
+* `id` - The ID of the Databricks Workspace in Azure Management plane.
```
Comment (tombuildsstuff) [inline website/docs/d/databricks_workspace.html.markdown:33]: we can simplify this to:

```suggestion
* `id` - The ID of the Databricks Workspace.
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @favoretti 

Thanks for this PR :)

Taking a look through this mostly LGTM - if we can fix up the comments then we should be able to run the tests and get this merged 👍 

Thanks!

**Comment by favoretti** [comment]
```
(⎈ |eastus2-fxs-oscar-np-fxc:build-agents)➜  terraform-providers/terraform-provider-azurerm git:(favoretti/databricks_workspace_datasource) $ make acctests SERVICE='databricks' TESTARGS='-run=TestAccDataSourceAzureRMDatabricksWorkspace' TESTTIMEOUT='60m'
==> Checking that code complies with gofmt requirements...
==> Checking that Custom Timeouts are used...
==> Checking that acceptance test packages are used...
TF_ACC=1 go test -v ./azurerm/internal/services/databricks/tests/ -run=TestAccDataSourceAzureRMDatabricksWorkspace -timeout 60m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccDataSourceAzureRMDatabricksWorkspace_basic
=== PAUSE TestAccDataSourceAzureRMDatabricksWorkspace_basic
=== CONT  TestAccDataSourceAzureRMDatabricksWorkspace_basic
--- PASS: TestAccDataSourceAzureRMDatabricksWorkspace_basic (569.69s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/databricks/tests	570.802s
```

**Review by katbyte — APPROVED** [review]
Thanks @favoretti - LGTM 👍

**Pull request merged**